src/pool-bin/BinPoolManager.sol

```diff
@@ -135,27 +136,19 @@ contract BinPoolManager is IBinPoolManager, ProtocolFees, Extsload {
         whenNotPaused
         returns (BalanceDelta delta)
     {
+        if (amountIn == 0) revert InsufficientAmountIn();
```

Comment:
This check has been moved from `BinPool.sol` to here otherwise the call will revert if `beforeSwap` has updated the actual `amountIn` to 0.